src/pages/library/common.js

```diff
@@ -161,34 +226,55 @@ export const SearchAndFilterComponent = (featuredList, sidebarSections, activeTa
     _.remove(section => _.isEmpty(section.labels))
   )(sidebarSections)
 
-  const sortWorkspaces = Utils.cond(
+  const sortData = Utils.cond(
     [sort === 'most recent', () => _.orderBy(['created'], ['desc'])],
     [sort === 'alphabetical', () => _.orderBy(w => _.toLower(_.trim(w.name)), ['asc'])],
+    [sort === 'Dataset Name', () => _.orderBy(w => _.toLower(_.trim(w.name)), [sortDir === 1 ? 'asc' : 'desc'])],
+    [sort === 'Project', () => _.orderBy(w => _.toLower(_.trim(w.project.name)), [sortDir === 1 ? 'asc' : 'desc'])],
+    [sort === 'No. of Subjects', () => _.orderBy(['subjects', 'lowerName'], [sortDir === 1 ? 'asc' : 'desc'])],
+    [sort === 'Data Type', () => _.orderBy(['dataType', 'lowerName'], [sortDir === 1 ? 'asc' : 'desc'])],
+    [sort === 'Last Updated', () => _.orderBy(['lastUpdated', 'lowerName'], [sortDir === 1 ? 'asc' : 'desc'])],
     () => _.identity
   )
 
-  const filterBySections = workspaces => {
+  const filterBySections = listdata => {
     if (_.isEmpty(selectedSections)) {
-      return workspaces
+      return listdata
     } else {
       const tags = _.uniq(_.flatMap(section => section.tags, selectedSections))
-      return _.uniq(_.flatMap(tag => workspacesByTag[tag], tags))
+      return _.uniq(_.flatMap(tag => listdataByTag[tag], tags))
     }
   }
   // eslint-disable-next-line lodash-fp/no-single-composition
-  const filterByTags = workspaces => _.flow(_.map(tag => _.intersection(workspacesByTag[tag]), selectedTags))(workspaces)
-  const filterByText = workspaces => {
+  const filterByTags = listdata => _.flow(_.map(tag => _.intersection(listdataByTag[tag]), selectedTags))(listdata)
+  const filterByText = listdata => {
     const lowerSearch = _.toLower(searchFilter)
     return _.isEmpty(lowerSearch) ?
-      workspaces :
-      _.filter(workspace => _.includes(lowerSearch, workspace.lowerName) || _.includes(lowerSearch, workspace.lowerDescription), workspaces)
+      listdata :
+      _.filter(listdatum => _.includes(lowerSearch, listdatum.lowerName) || _.includes(lowerSearch, listdatum.lowerDescription), listdata)
   }
-  const filteredWorkspaces = _.flow(
+  const filteredData = _.flow(
     filterBySections,
     filterByTags,
     filterByText
   )(featuredList)
 
+  const toggleData = data => {
+    if (_.some(data, selectedData)) {
```

Comment:
Please check order:
```suggestion
    if (_.includes(data, selectedData)) {
```

Reply:
When I switch to `_.includes`, the condition only ever returns as false - I don't understand the reasoning though, since according to the documentation, they should be pretty equivalent methods... For now, I'd like to propose we leave it as `_.some`

Reply:
Thank you for checking this. Indeed, `_.some` is much more powerful than I realized. While `_.includes` does an equality comparison, `_.some` will use different comparison mechanisms depending on the argument. I'm glad to have learned this.